packages/front-end/components/Experiment/SinglePage.tsx

```diff
@@ -485,7 +485,9 @@ export default function SinglePage({
           >
             <div className="appbox px-3 pt-3 pb-2">
               <RightRailSectionGroup title="Data Source" type="commaList">
-                {datasource?.name}
+                <Link href={`/datasources/${datasource.id}`}>
+                  {datasource?.name}
+                </Link>
```

Comment:
`datasource` may be undefined (select "Manual" as the data source).  If that's the case, this will throw an error as-is